docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md

```diff
@@ -0,0 +1,384 @@
+
+For more information, read the [Walkthrough: Run a pipeline.](../../walkthroughs/run-a-pipeline)
+
+## Sources and resources
+
+`dlt` works on the principle of [sources](../../general-usage/source) and
+[resources](../../general-usage/resource).
+
+### Source `mongodb`
+
+This function loads data from a MongoDB database, yielding collections to be retrieved.
+
+```python
+@dlt.source
+def mongodb(
+    connection_url: str = dlt.secrets.value,
+    database: Optional[str] = dlt.config.value,
+    collection_names: Optional[List[str]] = dlt.config.value,
+    incremental: Optional[dlt.sources.incremental] = None,  # type: ignore[type-arg]
+    write_disposition: Optional[str] = dlt.config.value,
+) -> Iterable[DltResource]:
+```
+
+Args:
+
+`connection_url`: MongoDB connection URL.
+
+`database`: Database name (defaults if unspecified).
+
+`collection_names`: Names of desired collections; loads all if not specified.
+
+`incremental`: Option for incremental data loading.
+
+`write_disposition`: Writing mode: "replace", "append", or "merge".
+
+Returns:
+
+`Iterable[DltResource]`: A list of DLT resources for each specified collection.
+
+### Resource `collection.name`
+
+This code segment is part of the source "mongodb", processes a list of MongoDB collections
+(collection_list). For each collection, a DLT resource is setup using "dlt.resource", with
+"collection_documents" as the data-yielding function.
+
+```python
+for collection in collection_list:
+    yield dlt.resource(  # type: ignore
+        collection_documents,
+        name=collection.name,
+        primary_key="_id",
+        write_disposition=write_disposition,
+        spec=MongoDbCollectionConfiguration,
+    )(client, collection, incremental=incremental)
+```
+
+`name`: The current collection's name.
+
+`primary_key`: '_id' is the default MongoDB primary key.
+
+`write_disposition`: Writing mode: "replace", "append", or "merge".
+
+The `spec` parameter deploys 'MongoDbCollectionConfiguration' for MongoDB setup. Each resource is
+invoked with the MongoDB client, current collection, and incremental parameter, preparing a DLT
+resource for each item in collection_list.
+
+### Source `mongo_collection`
+
+This function fetches data from a MongoDB collection, returning resources for each collection to be loaded.
+
+```python
+def mongodb_collection(
+    connection_url: str = dlt.secrets.value,
+    database: Optional[str] = dlt.config.value,
+    collection: str = dlt.config.value,
+    incremental: Optional[dlt.sources.incremental] = None,  # type: ignore[type-arg]
+    write_disposition: Optional[str] = dlt.config.value,
+) -> Any:
+```
+
+Args:
+
+The arguments `connection_url`, `database`, `incremental` and `write_disposition` are the same as
+defined for [Source `mongodb`](#source-mongodb).
+
+`collection`: Name of the collection to load.
+
+Returns:
+
+Iterable [DltResource]: A list of DLT resources ("collection_documents") for each collection to be
```

Comment:
it returns only one resource

Reply:
there is a mistake https://github.com/dlt-hub/verified-sources/blob/master/sources/mongodb/__init__.py#L83C1-L83C89

Reply:
corrected